source/commands/bounty.js

```diff
@@ -0,0 +1,65 @@
+const { PermissionFlagsBits, ActionRowBuilder, StringSelectMenuBuilder } = require('discord.js');
+const { CommandWrapper } = require('../classes');
+const { database } = require('../../database');
+const { getNumberEmoji } = require('../helpers');
+
+const customId = "bounty";
+const options = [];
+const subcommands = [
+	{
+		name: "post",
+		description: "Post your own bounty (+1 XP)",
+		optionsInput: []
+	}
+];
+module.exports = new CommandWrapper(customId, "Bounties are user-created objectives for other server members to complete", PermissionFlagsBits.ViewChannel, false, true, 3000, options, subcommands,
+	(interaction) => {
+		switch (interaction.options.getSubcommand()) {
+			case subcommands[0].name: // Post
```

Comment:
This looks a bit overengineered in the immediate term, but probably fine after adding a means to edit with this. Something probably better to mention though: it may be smart in the long term to have the functionality of each subcommand attached to it or otherwise put to a separate function, so that we can break out the code/functionality in a modular way.

Reply:
Planning to organize commands in UI with this structure (`/bounty post`, `/bounty edit`, `/bounty complete`, etc). See #3 for more info.

Totally fine with separating subcommands out into separate files or whatever since their responsibilities are going to be very different, but haven't gotten to it yet because that will also entail adapting the wiki generator script to be able to use those files.

Reply:
I'm more concerned about putting them in separate functions first, since their functionality shouldn't directly interact with each other outside of having the conceptual relation of `Bounty`, which this file already accomplishes. Personally, I'd prefer putting all of the related bounty "controller" code in the same place if possible, and putting them in separate functions may allow us to bundle all `bounty` code from different UI sources in the future.

Reply:
I'd be fine with that. A little skeptical about how frequently we'll actually get to reuse those functions with discord API tying interface to functionality, but if we can find some it'd be a win.

Reply:
It is much less about reuse and more about concentrating conceptually adjacent things. Having all `bounty` functionality in similar locations means that developers in the future have less overall preparation needed to do to work on `bounty` related functionality instead of bouncing between more contexts. In my opinion, bad context switching kills more of my time at work than many more serious issues, since it tends to happen incredibly frequently when it does occur.

Reply:
Although, to reiterate again from earlier: that kind of work would likely be something that should be a cleanup pass later on, and shouldn't be a reason to hold this back right now.